### ISSUE TITLE
Can't get xdebug working with platformsh recipe and PHPStorm

### ISSUE THREAD

**Issue by aimfeld** [lando/lando#2733]
I have tried quite a bit and read various sources, but I just can't figure out how to get xdebug working with lando and the platform.sh recipe. Otherwise, my local site runs fine. Any help is greatly appreciated. My OS is Windows 10.

Here are my settings:

.lando.yml
```
name: surveylab-blueprint
recipe: platformsh
config:
  id: bftge3smv4qgi
  xdebug: true
env_file:
  - .env
excludes:
  - .idea
  - .platform
  - .yarn
  - node_modules
  - vendor
  - "!vendor/cloud-solutions"
```

.platform.app.yaml
```
runtime:
  extensions:
    - redis
    - xdebug
  xdebug:
    idekey: PHPSTORM  # See https://docs.platform.sh/languages/php/xdebug.html

...
```

php.ini in /app 
```
; Xdebug
xdebug.remote_connect_back = 1
xdebug.idekey = "PHPSTORM"

...
```

PHP Settings:
![image](https://user-images.githubusercontent.com/760330/101347414-58b79280-388a-11eb-90cf-dacd53c22897.png)

CLI Interpreter:
![image](https://user-images.githubusercontent.com/760330/101347481-74bb3400-388a-11eb-8a19-df04c7bb8a13.png)

Debug Settings:
![image](https://user-images.githubusercontent.com/760330/101347583-94eaf300-388a-11eb-875e-6000ebf82c0b.png)

DBGp Proxy: 
![image](https://user-images.githubusercontent.com/760330/101347637-acc27700-388a-11eb-8002-9ab3c19eb896.png)

PHP Info:
![image](https://user-images.githubusercontent.com/760330/101347811-e8f5d780-388a-11eb-8201-9df38a9322f3.png)
![image](https://user-images.githubusercontent.com/760330/101348001-370adb00-388b-11eb-8440-b09b74a1445d.png)
![image](https://user-images.githubusercontent.com/760330/101348040-4c800500-388b-11eb-9428-8f04e8c5cd30.png)
![image](https://user-images.githubusercontent.com/760330/101348083-60c40200-388b-11eb-8de6-7f9e6c794857.png)

**Comment by pirog** [comment]
@labboy0276 you've gotten this to work right?

**Comment by labboy0276** [comment]
@pirog what @aimfeld has there seems correct.  I will take some time in this week and see if I can replicate and report back with a fix if needed.

**Comment by Adstokoe19** [comment]
@aimfeld if you are using Xdebug 3, you need to update your settings as per https://xdebug.org/docs/upgrade_guide for example, in your lando.yml file,` xdebug: true` is now `xdebug: debug`.

**Comment by pirog** [comment]
Platform.sh provides its own images so the xdebug configuration is likely
different eg you would set it up according to their docs

**Comment by labboy0276** [comment]
@aimfeld After looking at one of my sites, you will need to change your files to the following:

.lando.yml
```
name: surveylab-blueprint
recipe: platformsh
config:
  id: bftge3smv4qgi
  variables:
    app:
      php:
        xdebug.remote_connect_back: 1
env_file:
  - .env
excludes:
  - .idea
  - .platform
  - .yarn
  - node_modules
  - vendor
  - "!vendor/cloud-solutions"
```

.platform.app.yaml
```
runtime:
  extensions:
    - redis
    - xdebug
```

Remove the IDEKEY from PHPStorm as well.  Also, delete your php.ini.  The p.sh documents aren't exactly on par with how things work in lando.  I will update the docs here: https://docs.lando.dev/config/platformsh.html#xdebug @pirog 

If you are still having issue @aimfeld , I suggest getting the chrome xdebug browser extension and turning that on.

**Comment by pirog** [comment]
@labboy0276 updated the docs with your suggestion

**Comment by aimfeld** [comment]
@labboy0276 Thanks for the quick help! In the meantime, I set up a whole new OS (Linux Mint) with a fresh install of PHPStorm, Docker, and Lando. 

Now I got xdebug 2.9.8 working with very few settings:

.lando.yml: No xdebug-related settings needed at all. Setting `xdebug.remote_connect_back: 1` here did not work for me. It didn't show up as enabled in phpinfo either.
```
name: surveylab-blueprint
recipe: platformsh
config:
  id: bftge3smv4qgi
env_file:
  - .env
excludes:
...
```

.platform.app.yaml: removed xdebug idekey, as you suggested
```
runtime:
  extensions:
    - redis
    - xdebug
...
```

php.ini in ` /app`: These settings are both necessary. Removing either and then `lando restart` will make xdebug not work.
```
; Xdebug
xdebug.remote_connect_back = 1
xdebug.idekey = "PHPSTORM"

...
```
